oneview/resource_logical_interconnect_group.go

```diff
@@ -597,9 +775,159 @@ func resourceLogicalInterconnectGroupCreate(d *schema.ResourceData, meta interfa
+			sflowCollector.MaxDatagramSize = val.(int)
+		}
+
+		if val, ok := d.GetOk(sflowCollectorPrefix + ".max_header_size"); ok {
+			sflowCollector.MaxHeaderSize = val.(int)
+		}
+
+		if val, ok := d.GetOk(sflowCollectorPrefix + ".name"); ok {
+			sflowCollector.Name = val.(string)
+		}
+
+		if val, ok := d.GetOk(sflowCollectorPrefix + ".port"); ok {
+			sflowCollector.Port = val.(int)
+		}
+
+		sflowCollectors = append(sflowCollectors, sflowCollector)
+	}
+	sflowConfiguration.SflowCollectors = sflowCollectors
+
+	sflowNetworkPrefix := fmt.Sprintf(sflowConfigurationPrefix + ".sflow_network.0")
+	sflowNetwork := ov.SflowNetwork{}
+
+	if val, ok := d.GetOk(sflowNetworkPrefix + ".vlan_id"); ok {
+		sflowNetwork.VlanId = val.(int)
+	}
+
+	if val, ok := d.GetOk(sflowNetworkPrefix + ".uri"); ok {
+		sflowNetwork.URI = utils.Nstring(val.(string))
+	}
+
+	if val, ok := d.GetOk(sflowNetworkPrefix + ".name"); ok {
+		sflowNetwork.Name = val.(string)
+	}
+
+	sflowConfiguration.SflowNetwork = &sflowNetwork
+
+	sflowPortCount := d.Get(sflowConfigurationPrefix + ".sflow_ports.#").(int)
+	sflowPorts := make([]ov.SflowPort, 0)
+	for i := 0; i < sflowPortCount; i++ {
+		sflowPortPrefix := fmt.Sprintf(sflowConfigurationPrefix+".sflow_ports.%d", i)
+		sflowPort := ov.SflowPort{}
+
+		if val, ok := d.GetOk(sflowPortPrefix + ".bay_number"); ok {
+			sflowPort.BayNumber = val.(int)
+		}
+
+		if val, ok := d.GetOk(sflowPortPrefix + ".enclosure_index"); ok {
+			sflowPort.EnclosureIndex = val.(int)
+		}
+
+		if val, ok := d.GetOk(sflowPortPrefix + ".collector_id"); ok {
+			sflowPort.CollectorId = val.(int)
+		}
+
+		if val, ok := d.GetOk(sflowPortPrefix + ".icm_name"); ok {
+			sflowPort.IcmName = val.(string)
+		}
+
+		if val, ok := d.GetOk(sflowPortPrefix + ".port_num"); ok {
+			sflowPort.PortName = val.(string)
+		}
+
+		sflowPorts = append(sflowPorts, sflowPort)
+	}
+	sflowConfiguration.SflowPorts = sflowPorts
+
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".category"); ok {
+		sflowConfiguration.Category = val.(string)
+	}
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".description"); ok {
+		sflowConfiguration.Description = utils.NewNstring(val.(string))
+	}
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".enabled"); ok {
+		enabled := val.(bool)
+		sflowConfiguration.Enabled = &enabled
+	}
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".name"); ok {
+		sflowConfiguration.Name = val.(string)
+	}
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".state"); ok {
+		sflowConfiguration.State = val.(string)
+	}
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".status"); ok {
+		sflowConfiguration.Status = val.(string)
+	}
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".uri"); ok {
+		sflowConfiguration.URI = utils.NewNstring(val.(string))
+	}
+
+	if val, ok := d.GetOk(sflowConfigurationPrefix + ".type"); ok {
+		sflowConfiguration.Type = val.(string)
+		lig.SflowConfiguration = &sflowConfiguration
```

Comment:
Type being an optional attribute, Is the non-assignment in line928 expected if type value is not provided?

Reply:
i gave type as optional so the value can be empty. It is mandatory if this struct is passed.